### PR TITLE
[receiver-mock] Split into files

### PR DESCRIPTION
**Pull request by pmalek-sumo**

### CODE REVIEW

File: src/rust/receiver-mock/src/main.rs [hunk 1]
```diff
@@ -288,21 +45,44 @@ pub async fn main() {
...
+        ));
+        Ok::<_, Infallible>(result)
+    }});
+
+    let server = Server::bind(&addr).serve(make_svc);
+
+    if let Err(e) = server.await {
+        eprintln!("server error: {}", e);
+    }
 }
```
Comment (sumo-drosiek) [inline src/rust/receiver-mock/src/main.rs:88]: I would add newline to be consistent

File: src/rust/receiver-mock/src/metrics.rs [hunk 2]
```diff
@@ -0,0 +1,47 @@
+use std::net::IpAddr;
+use std::sync::{Arc, Mutex};
+
+pub fn handle_carbon2(
+    lines: std::str::Lines,
+    address: IpAddr,
+    stats: &Arc<Mutex<crate::statistics::Statistics>>,
```
Comment (sumo-drosiek) [inline src/rust/receiver-mock/src/metrics.rs:7]: I would rather use `crate::statistics::Statistics` and `Statistics`

Comment (pmalek-sumo) [reply src/rust/receiver-mock/src/metrics.rs]: done

File: src/rust/receiver-mock/src/main.rs (hunk 1 above)
Comment (pmalek-sumo) [reply src/rust/receiver-mock/src/main.rs]: done

File: src/rust/receiver-mock/src/main.rs [hunk 3]
```diff
@@ -1,259 +1,17 @@
...
-      (*statistics).ts = get_now();
-      (*statistics).p_metrics = (*statistics).metrics;
-      (*statistics).p_logs = (*statistics).logs;
-      (*statistics).p_logs_bytes = (*statistics).logs_bytes;
-  }
-}
+mod metrics;
+mod router;
+mod statistics;
+use statistics::Statistics;
```
Comment (pmalek-sumo) [inline src/rust/receiver-mock/src/main.rs:13]: I'd like to know:

* why do I need both of those here? When I skip `mod statistics` I get
  ```
  unresolved import `statistics`

  use of undeclared type or module `statistics`rustc(E0432)
  ```
* I can use `use crate::statistics;` in e.g. `router.rs` but in `main.rs` when I use it I get
  ```
  unresolved import `crate::statistics`

  no `statistics` in the rootrustc(E0432)
  ```

@sumo-drosiek any ideas?

Comment (sumo-drosiek) [reply src/rust/receiver-mock/src/main.rs:13]: I put both `mod` and `use` and cargo returns this:

```
error[E0255]: the name `statistics` is defined multiple times
  --> src/main.rs:15:5
   |
12 | mod statistics;
   | --------------- previous definition of the module `statistics` here
...
15 | use crate::statistics;
   |     ^^^^^^^^^^^^^^^^^ `statistics` reimported here
   |
   = note: `statistics` must be defined only once in the type namespace of this module
help: you can use `as` to change the binding name of the import
   |
15 | use crate::statistics as other_statistics;
```

You need to declare `mod statistics` somewhere, so you can use `crate::statistics` in other files, and you cannot overwrite module by importing another one in it's place

**Pull request merged**